librosa/core/constantq.py

```diff
@@ -612,115 +604,80 @@ def icqt(C, sr=22050, hop_length=512, fmin=None,
     >>> y_hat = librosa.icqt(C=C, sr=sr, hop_length=hop_length,
     ...                 bins_per_octave=bins_per_octave)
     '''
-    warnings.warn('librosa.icqt is unstable, and subject to change in future versions. '
-                  'Please use with caution.')
-
-    n_bins, n_frames = C.shape
-    n_octaves = int(np.ceil(float(n_bins) / bins_per_octave))
-
-    if amin is None:
-        amin = util.tiny(C)
 
     if fmin is None:
         fmin = note_to_hz('C1')
 
-    freqs = cqt_frequencies(n_bins,
-                            fmin,
+    # Get the top octave of frequencies
+    n_bins = len(C)
+
+    freqs = cqt_frequencies(n_bins, fmin,
                             bins_per_octave=bins_per_octave,
                             tuning=tuning)[-bins_per_octave:]
 
-    fmin_t = np.min(freqs)
-
-    # Make the filter bank
-    basis, lengths = filters.constant_q(sr=sr,
-                                        fmin=fmin_t,
-                                        n_bins=bins_per_octave,
-                                        bins_per_octave=bins_per_octave,
-                                        filter_scale=filter_scale,
-                                        tuning=tuning,
-                                        norm=norm,
-                                        window=window,
-                                        pad_fft=True)
-    n_fft = basis.shape[1]
+    fmin_t = np.min(freqs[-bins_per_octave:])
 
-    # The extra factor of lengths**0.5 corrects for within-octave tapering
-    basis = basis * np.sqrt(lengths[:, np.newaxis])
+    n_filters = min(n_bins, bins_per_octave)
 
-    # Estimate the gain per filter
-    bdot = basis.conj().dot(basis.T)
-    bscale = np.sum(np.abs(bdot), axis=1)
+    fft_basis, n_fft, lengths = __cqt_filter_fft(sr, fmin_t,
+                                                 n_filters,
+                                                 bins_per_octave,
+                                                 tuning,
+                                                 filter_scale,
+                                                 norm,
+                                                 sparsity=sparsity,
+                                                 window=window)
 
-    n_trim = basis.shape[1] // 2
+    if min(lengths) < hop_length:
+        warnings.warn('Minimum CQT filter length={:.3f} is less than hop_length={}.\n'
```

Comment:
I would phrase it the other way around: `hop length is greater than minimum CQT filter length`
because the most common fix is to lower the hop length.